addons/stock_account/models/product.py

```diff
@@ -716,6 +716,8 @@ def _svl_replenish_stock(self, description, products_orig_quantity_svl):
                     qty_to_remove = min(qty, quantity_svl)
                     quantity_svl -= qty_to_remove
                     svl_vals = product._prepare_in_svl_vals(qty_to_remove, price_unit, lot=lot)
+                    if lot:
```

Comment:
Should be done outside of the `for` loop on all lots at the same time (as it's the same value).